simplivity/resources/policies.py

```diff
@@ -91,11 +91,11 @@ def get_vms(self):
         method_url = "{}/{}/virtual_machines".format(URL, self.data["id"])
         vm_data = self._client.do_get(method_url).get("virtual_machines", [])
 
-        vms_obj = resources.virtual_machines.VirtualMachines(self._connection)
+        vms_obj = virtual_machines.VirtualMachines(self._connection)
         vms = []
 
         for vm in vm_data:
-            vms.append(vms_obj.get_by_id(vm["id"]))
+            vms.append(vms_obj.get_by_data(vm))
```

Comment:
This was change from get_by_id to get_by_data to prevent having to go out to the rest server multiple time for a single call.

I was able to use a simple script that was able to get all vm(s) from a given policy, and create a backup using the vm object that was returned by this method.

i.e
all_policies = policies.get_all()

for vm in all_policies[0].get_vms():
    vm.create_backup(backup_name=time.time())
    print(vm.get_backups())

Reply:
Agree with you, this way we can avoid hitting the server unwantedly.